helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -77,7 +88,23 @@ public synchronized boolean addMetadataStoreRealm(String realm) {
       return createZkRealm(realm);
     }
 
-    // TODO: Forward the request to leader
+    String leaderHostName = _leaderElection.getCurrentLeaderInfo().getId();
+    String url =
+        leaderHostName + "/admin/v2/namespaces/" + _namespace + "/metadata-store-realms/" + realm;
```

Comment:
Agreed.

Reply:
`_leaderHostName` is an instance variable now since it's used during multiple occasions. I'm refactoring the calling code since it can be generalized and reduce duplication.

Reply:
On a second thought I cannot do that because `_leaderHostName` will be changing. Modified my planned code structure for that. Duplication can still be reduced.